pkg/reconciler/githubsource/webhook_client.go

```diff
@@ -38,14 +39,29 @@ type webhookOptions struct {
 type webhookClient interface {
 	Create(ctx context.Context, options *webhookOptions) (string, error)
 	Delete(ctx context.Context, options *webhookOptions, hookID string) error
+	CreateWithGitHubBaseURL(ctx context.Context, options *webhookOptions, alternateGitHubAPIURL string) (string, error)
+	DeleteWithGitHubBaseURL(ctx context.Context, options *webhookOptions, hookID string, alternateGitHubAPIURL string) error
 }
 
 type gitHubWebhookClient struct{}
 
 func (client gitHubWebhookClient) Create(ctx context.Context, options *webhookOptions) (string, error) {
+	return client.CreateWithGitHubBaseURL(ctx, options, "")
+}
+
+func (client gitHubWebhookClient) Delete(ctx context.Context, options *webhookOptions, hookID string) error {
+	return client.DeleteWithGitHubBaseURL(ctx, options, hookID, "")
+}
+
+func (client gitHubWebhookClient) CreateWithGitHubBaseURL(ctx context.Context, options *webhookOptions, alternateGitHubAPIURL string) (string, error) {
 	logger := logging.FromContext(ctx)
 
 	ghClient := client.createGitHubClient(ctx, options)
+	if alternateGitHubAPIURL != "" {
+		//This is to support GitHub Enterprise, this might be something like https://github.company.com/api/v3/
+		ghClient.BaseURL, _ = url.Parse(alternateGitHubAPIURL)
```

Comment:
Handle the error (might be as simple as returning it).

Reply:
Good spot (pesky underscores) - error now returned